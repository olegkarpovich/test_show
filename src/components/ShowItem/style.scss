#show-general-information {
  width: 100%;
  display: flex;
  flex-direction: row;
  align-items: top;

  #show-main-image {
    float: left;

    img {
      max-width: 100%;
    }
  }

  #show-description {
    width: 100%;
    display: flex;
    flex-direction: column;
    align-items: center;
    text-align: left;

    .show-data-blocks {
      width: 100%;
      display: flex;
      flex-direction: row;
      align-items: top;
      justify-content: center;

      #show-data-blocks-language {
        margin-right: 10px;
      }
    }
  }
}

@media(max-width: 768px) {
  #show-description {
    font-size: 12px;
  }
}

@media(max-width: 480px) {
  #show-general-information {
    flex-direction: column !important;

    .ep-block-header {
      text-transform: none;
    }

    #show-description {
      font-size: 10px;
      text-align: center !important;

      p {
        display: block;
        width: 95% !important;
      }
    }
  }
}






