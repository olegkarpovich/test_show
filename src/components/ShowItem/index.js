import React, {Component} from 'react'
import './style.scss';

function ShowItem(props) {
    const {show, children} = props;
    return (
        <section id="show-general-information">
            <aside id="show-main-image">
                <figure>
                    {show.image && <img src={show.image.medium} alt={show.name}/>}
                </figure>
            </aside>
            <article id="show-description">
                <h3>{show.name}</h3>
                <section className="show-data-blocks">
                    <article id="show-data-blocks-language"> {`Language: ${show.language}`} </article>
                    <article id="show-data-blocks-raiting"> {`Rating: ${show.rating && show.rating.average}`} </article>
                </section>
                <div dangerouslySetInnerHTML={{__html: show.summary}}/>
                {children}
            </article>
        </section>
    )
}

export default ShowItem;
