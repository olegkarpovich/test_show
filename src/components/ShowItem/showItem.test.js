import React from 'react';
import { render, act } from '@testing-library/react';
import ShowItem from './index';

const episode = {
    show: {
        image: {
            medium: 'static.tvmaze.com/uploads/images/medium_landscape/53/132617.jpg'
        },
        name: 'Escape from Monster Island',
        summary: 'Bubbles wins two tickets to a concert and has to pick between Blossom and Buttercup to go with her. Meanwhile, the Mayor\'s plane goes down over Monster Island and it\'s up to the girls to rescue him.',
        language: 'English',
        rating: {
            average: 6
        }
    }
};

const component = () => (
    <ShowItem {...episode}/>
);

describe('ShowItem component', () => {
    it('should render show details page', () => {
        const { container } = render(component());

        expect(container).toMatchSnapshot();
    });
});