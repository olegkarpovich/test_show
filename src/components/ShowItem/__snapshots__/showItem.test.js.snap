// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShowItem component should render show details page 1`] = `
<div>
  <section
    id="show-general-information"
  >
    <aside
      id="show-main-image"
    >
      <figure>
        <img
          alt="Escape from Monster Island"
          src="static.tvmaze.com/uploads/images/medium_landscape/53/132617.jpg"
        />
      </figure>
    </aside>
    <article
      id="show-description"
    >
      <h3>
        Escape from Monster Island
      </h3>
      <section
        class="show-data-blocks"
      >
        <article
          id="show-data-blocks-language"
        >
           
          Language: English
           
        </article>
        <article
          id="show-data-blocks-raiting"
        >
           
          Rating: 6
           
        </article>
      </section>
      <div>
        Bubbles wins two tickets to a concert and has to pick between Blossom and Buttercup to go with her. Meanwhile, the Mayor's plane goes down over Monster Island and it's up to the girls to rescue him.
      </div>
    </article>
  </section>
</div>
`;
